doc/how-to/vshard_quick.rst

```diff
@@ -1,160 +1,525 @@
+        :end-at: 127.0.0.1:3300
+        :dedent:
+
+    The main group-level options here are:
+
+    *   ``app``: The ``app.module`` option specifies that code specific to a router should be loaded from the ``router`` module. This is explained below in the :ref:`vshard-quick-start-router-code` section.
+    *   ``sharding``: The ``sharding.roles`` option specifies that an instance inside this group acts as a router.
+    *   ``replicasets``: This section configures one replica set with one router instance.
+
+
+Resulting configuration
+***********************
+
+The resulting cluster configuration should look as follows:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :dedent:
+
+
+..  _vshard-quick-start-storage-code:
+
+Adding storage code
+~~~~~~~~~~~~~~~~~~~
+
+1.  Open the ``storage.lua`` file and create a space using the :ref:`box.schema.space.create() <box_schema-space_create>` function:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/storage.lua
+        :language: lua
+        :start-at: box.schema.create_space
+        :end-before: box.space.bands:create_index('id'
+        :dedent:
+
+    Note that the created ``bands`` spaces includes the ``bucket_id`` field.
+    This field represents a sharding key used to partition a dataset across different storage instances.
+
+2.  Create two indexes based on the ``id`` and ``bucket_id`` fields:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/storage.lua
+        :language: lua
+        :start-at: box.space.bands:create_index('id'
+        :end-at: box.space.bands:create_index('bucket_id'
+        :dedent:
+
+3.  Define the ``insert_band`` function that inserts a tuple into the created space:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/storage.lua
+        :language: lua
+        :start-at: function insert_band
+        :end-before: function get_band
+        :dedent:
+
+4.  Define the ``get_band`` function that returns data without the ``bucket_id`` value:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/storage.lua
+        :language: lua
+        :start-at: function get_band
+        :dedent:
+
+The resulting ``storage.lua`` file should look as follows:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/storage.lua
+    :language: lua
+    :dedent:
+
+
+..  _vshard-quick-start-router-code:
+
+Adding router code
+~~~~~~~~~~~~~~~~~~
+
+1.  Open the ``router.lua`` file and load the ``vshard`` module as follows:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/router.lua
+        :language: lua
+        :start-at: local vshard
+        :end-at: local vshard
+        :dedent:
+
+2.  Define the ``put`` function used to write data to a storage:
```

Comment:
I'd like more context here: say explicitly that this function defines how the router selects the storage to write the data. And the same for reading.

Reply:
> this function defines how the router selects the storage to write the data

This function does two things:
1) Calculates a bucket ID used to write data to the correct storage.
2) Uses this bucket ID to write data to the storage.

This is what the list after the function definition describes. I'd keep the first sentence short because too much context before the function definition makes it hard to read because of too much technical details.